acme/dns_challenge.go

```diff
@@ -203,13 +215,13 @@ func FindZoneByFqdn(fqdn, nameserver string) (string, error) {
 	//   Name servers authoritative for a zone MUST include the SOA record of
 	//   the zone in the authority section of the response when reporting an
 	//   NXDOMAIN or indicating that no data (NODATA) of the requested type exists
-	in, err := dnsQuery(fqdn, dns.TypeSOA, nameserver, true)
+	in, err := dnsQuery(fqdn, dns.TypeSOA, nameservers, true)
 	if err != nil {
 		return "", err
 	}
 	if in.Rcode != dns.RcodeNameError {
 		if in.Rcode != dns.RcodeSuccess {
-			return "", fmt.Errorf("NS %s returned %s for %s", nameserver, dns.RcodeToString[in.Rcode], fqdn)
+			return "", fmt.Errorf("The NS returned %s for %s", dns.RcodeToString[in.Rcode], fqdn)
```

Comment:
If dnsQuery were to return the nameserver it used to make the query, we could allow the error to be more informative again. Do we consider this as issue or not?

For the sake of committing something, I've gone with not specifically mentioning the nameserver here and [below](https://github.com/xenolf/lego/pull/186/files#diff-3dd431e0b068070def9558d0daeef06cR240)